design/datamover.md

```diff
@@ -0,0 +1,208 @@
+# Add Datamover to Velero
+## Abstract
+As of today, Velero supports CSI snapshots. But there is a missing gap in functionality, the CSI snapshots are not durable as well as portable.
+It is desired that Velero support a concept of Datamover. We would like to propose an extensible design to support Datamovers, where-in the users
+can bring their own datamover implementation and use it to replicate CSI snapshots to object storage during backup and also recreate PVCs from the snapshots
+in object storage during restore. Thus making the CSI snapshots durable as well as portable.
+
+## Background
+Currently, Velero has two option to support backing up of persistent volumes.
+
+1. Restic: This solution performs filesystem level backups and mostly used when a volume snapshot plugin is not available. However, the restic integration is
+   tightly coupled with Velero, makes it difficult for someone to bring their own implementation of data movement. Although restic is portable but it is not
+   crash consistent as it requires quiescing of data before the backup is taken thus affecting the availability of application.
+
+2. Volume Snapshots: Storage providers provide with volume snapshot capabilities via plugins. These snapshots are crash consistent but there is no guarantee that
+   these snapshots are durable. These snapshots are generally stored locally in the cluster storage, so if the cluster breaks then the snapshots are of no use. Velero also supports
+   CSI volume snapshots, but as there is no data movement provided by Velero for these CSI snapshots they are not durable and dependent on the storage provider.
+
+
+Related Issue: https://github.com/vmware-tanzu/velero/issues/3229
+
+## Goals
+- To be able to backup CSI snapshots to object storage
+- To be able to recreate CSI snapshots under a different storage provider
+- To provide an extensible data mover solution which should allow third parties to plugin their own datamover implementation
+- To provide a Datamover example PoC
+
```

Comment:
Apart from the example PoC, it looks like this design has two major goals:

1)   export/import CSI snapshot to/from S3;
2)   extensible data mover solution. 

From my opinion, these are two different topics, but it seems we are trying to put them together into one design, by implementing 2, we can easily achieve 1. However, from this document, it looks like it's more focused on 2, but lack some consideration for 1.

For example, for goal 1, the design might also include retention and frequency management where user can setup the export differently from  backup. E.g., a user may want to do CSI snapshot backup hourly, then export one snapshot daily; and only keep 7 snapshots per PV in a week, then want to keep 1 monthly export.  Basically retention and frequency are common configurations from traditional backup softwares.